utils/nctl/sh/assets/setup_shared.sh

```diff
@@ -457,20 +489,34 @@ function setup_asset_global_state_toml() {
             if [ -f "$PATH_TO_NET/nodes/node-$IDX/storage/$(get_chain_name)/data.lmdb" ]; then
                 GLOBAL_STATE_OUTPUT=$("$NCTL_CASPER_HOME"/target/"$NCTL_COMPILE_TARGET"/global-state-update-gen \
                         system-contract-registry -d "$PATH_TO_NET"/nodes/node-"$IDX"/storage/"$(get_chain_name)" -s "$(nctl-view-chain-state-root-hash node=$IDX | awk '{ print $12 }' | tr '[:upper:]' '[:lower:]')")
+
+                GLOBAL_STATE_VALIDATOR_OUTPUT=$("$NCTL_CASPER_HOME"/target/"$NCTL_COMPILE_TARGET"/global-state-update-gen \
+                        validators $VALIDATOR_SETUP_STRING -d "$PATH_TO_NET"/nodes/node-"$IDX"/storage/"$(get_chain_name)" -s "$(nctl-view-chain-state-root-hash node=$IDX | awk '{ print $12 }' | tr '[:upper:]' '[:lower:]')")
             else
                 GLOBAL_STATE_OUTPUT=$("$NCTL_CASPER_HOME"/target/"$NCTL_COMPILE_TARGET"/global-state-update-gen \
                         system-contract-registry -d "$PATH_TO_NET"/nodes/node-1/storage/"$(get_chain_name)" -s "$(nctl-view-chain-state-root-hash node=1 | awk '{ print $12 }' | tr '[:upper:]' '[:lower:]')")
+
+                GLOBAL_STATE_VALIDATOR_OUTPUT=$("$NCTL_CASPER_HOME"/target/"$NCTL_COMPILE_TARGET"/global-state-update-gen \
+                        validators $VALIDATOR_SETUP_STRING -d "$PATH_TO_NET"/nodes/node-1/storage/"$(get_chain_name)" -s "$(nctl-view-chain-state-root-hash node=1 | awk '{ print $12 }' | tr '[:upper:]' '[:lower:]')")
             fi
         else
             if [ -f "$PATH_TO_NET/nodes/node-$IDX/storage/data.lmdb" ]; then
                 GLOBAL_STATE_OUTPUT=$("$NCTL_CASPER_HOME"/target/"$NCTL_COMPILE_TARGET"/global-state-update-gen \
                         system-contract-registry -d "$PATH_TO_NET"/nodes/node-"$IDX"/storage)
+
+                GLOBAL_STATE_VALIDATOR_OUTPUT=$("$NCTL_CASPER_HOME"/target/"$NCTL_COMPILE_TARGET"/global-state-update-gen \
+                        validators $VALIDATOR_SETUP_STRING -d "$PATH_TO_NET"/nodes/node-"$IDX"/storage)
+
             else
                 GLOBAL_STATE_OUTPUT=$("$NCTL_CASPER_HOME"/target/"$NCTL_COMPILE_TARGET"/global-state-update-gen \
                         system-contract-registry -d "$PATH_TO_NET"/nodes/node-1/storage)
+
+                GLOBAL_STATE_VALIDATOR_OUTPUT=$("$NCTL_CASPER_HOME"/target/"$NCTL_COMPILE_TARGET"/global-state-update-gen \
+                        validators $VALIDATOR_SETUP_STRING -d "$PATH_TO_NET"/nodes/node-1/storage)
```

Comment:
Note to self: We could use some discussion whether `system-contract-registry` is still needed (and if it is, maybe it should generate `validators` fields as well). To be discussed
cc: @fizyk20

Reply:
Added the `do not merge` label, because it was agreed that we need to change the following:

1. Rename the `validators` subcommand of `global-state-update-gen` to `change-validators`
2. Trigger the `system-contract-registry` only when upgrading from pre 1.4.0 version. In such case, we also need to invoke the `change-validators` becasue if the `global_state.toml` is created it **must** contain the `validators` entry
3.  When upgrading from version 1.4.0 or later, we should not invoke either the `system-contract-registry` or the `change-validators` subcommand (upgrade should work without a `global_state.toml` file

Also, we need to introduce an upgrade test that twiddles validator set (separate ticket).

Reply:
PR is ready for review again.

Reply:
rename `system-contract-registry` to `migrate-into-system-contract-registry` please

Reply:
Done in https://github.com/casper-network/casper-node/pull/3451/commits/9e502804931d5ef5c11f381952eafeb46b8095ac